cg/meta/workflow/prepare_fastq.py

```diff
@@ -26,16 +27,30 @@ def __init__(self, store: Store, compress_api: CompressAPI):
 
     def get_compression_objects(self, case_id: str) -> List[CompressionData]:
         """Return a list of compression objects"""
-        case_obj: Family = self.store.get_case_by_internal_id(internal_id=case_id)
+        case: Family = self.store.get_case_by_internal_id(internal_id=case_id)
         compression_objects = []
-        for link in case_obj.links:
-            sample_id = link.sample.internal_id
+        for link in case.links:
+            sample: Sample = link.sample
+            sample_id = sample.internal_id
+            if self._should_skip_sample(case=case, sample=sample):
+                LOG.warning(f"Skipping sample {sample_id} - it has no reads.")
+                continue
             version_obj: Version = self.compress_api.hk_api.get_latest_bundle_version(
                 bundle_name=sample_id
             )
             compression_objects.extend(files.get_spring_paths(version_obj))
         return compression_objects
 
+    @staticmethod
+    def _should_skip_sample(case: Family, sample: Sample):
+        """
+        For some pipelines, we want to start a partial analysis disregarding the samples with no reads.
+        This method returns true if we should skip the sample.
+        """
+        if case.data_analysis in [Pipeline.MICROSALT, Pipeline.SARS_COV_2] and not sample.has_reads:
```

Comment:
Maybe `[Pipeline.MICROSALT, Pipeline.SARS_COV_2]` can be a constant in a `constants.py` file? It might make it easier to understand when refactoring

Reply:
Done!